src/overlays/actors/ovl_Obj_Spidertent/z_obj_spidertent.c

```diff
@@ -28,118 +35,789 @@ const ActorInit Obj_Spidertent_InitVars = {
+    /* 0x24 */ f32 unk_24;
+} _struct_D_80B31350_0x28;
+
+_struct_D_80B31350_0x28 D_80B31350[] = {
     {
-        { ELEMTYPE_UNK4, { 0x00000000, 0x00, 0x00 }, { 0x00000800, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
-        { { { 57.20000076293945f, 0.0f, -47.79999923706055f }, { 17.799999237060547f, 0.0f, -87.19999694824219f }, { 0.8999999761581421f, 30.0f, -2.799999952316284f } } },
+        &object_spidertent_DL_000070,
+        &object_spidertent_Colheader_0011AC,
+        &sTrisInit1,
+        0.9f,
+        1.3f,
+        40.0f,
+        -3.8f,
+        14400.0f,
+        240.0f,
+        200.0f,
     },
     {
-        { ELEMTYPE_UNK4, { 0x00000000, 0x00, 0x00 }, { 0x00000800, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
-        { { { 51.599998474121094f, 0.0f, 14.100000381469727f }, { 57.20000076293945f, 0.0f, -47.79999923706055f }, { 0.8999999761581421f, 30.0f, -2.799999952316284f } } },
+        &object_spidertent_DL_001250,
+        &object_spidertent_Colheader_00238C,
+        &sTrisInit2,
+        0.9f,
+        0.9f,
+        30.0f,
+        -2.8f,
+        7744.0f,
+        176.0f,
+        168.0f,
     },
 };
 
-// static ColliderTrisInit sTrisInit = {
-static ColliderTrisInit D_80B31340 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_TRIS, },
-    6, D_80B311D8, // sTrisElementsInit,
+typedef struct {
+    /* 0x00 */ f32 unk_00;
+    /* 0x04 */ f32 unk_04;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ s16 unk_0C;
+    /* 0x0E */ s8 unk_0E;
+    /* 0x0F */ s8 unk_0F;
+    /* 0x10 */ Color_RGBA8 unk_10;
+    /* 0x14 */ Color_RGBA8 unk_14;
+} _struct_D_80B313A0_0x18;
+
+_struct_D_80B313A0_0x18 D_80B313A0[] = {
+    {
+        -20.0f,
+        6.0f,
+        0.0f,
+        100,
+        14,
+        5,
+        { 255, 255, 160, 160 },
+        { 255, 0, 0, 0 },
+    },
+    {
+        0.0f,
+        4.0f,
+        1.2f,
+        80,
+        9,
+        3,
+        { 255, 255, 160, 120 },
+        { 255, 0, 0, 0 },
+    },
+    {
+        0.0f,
+        4.2f,
+        0.3f,
+        70,
+        12,
+        2,
+        { 255, 255, 160, 100 },
+        { 55, 0, 0, 0 },
+    },
+    {
+        0.0f,
+        3.0f,
+        0.0f,
+        20,
+        6,
+        2,
+        { 255, 100, 20, 80 },
+        { 255, 0, 0, 0 },
+    },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B31418[] = {
+Vec3f D_80B31400 = { 1.0f, 0.0f, 0.0f };
+Vec3f D_80B3140C = { 0.0f, 1.0f, 0.0f };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
 };
 
-#endif
+s32 func_80B2FB10(Vec3f* arg0, Vec3f* arg1) {
+    f32 temp_f0 = Math3D_LengthSquared(arg0);
+    f32 temp_f2;
+
+    if (temp_f0 < 9.999999e-9f) {
```

Comment:
```suggestion
    if (temp_f0 < (SQ(1.0e-5) - 1)) {
```
This is probably better?

Reply:
Doesn't work.